packages/zowe-explorer-ftp-extension/src/ZoweExplorerFtpJesApi.ts

```diff
@@ -0,0 +1,249 @@
+        }
+        return;
+    }
+
+    public async getSpoolContentById(jobname: string, jobid: string, spoolId: number): Promise<string> {
+        const connection = await this.ftpClient(this.checkedProfile());
+        if (connection) {
+            const options = {
+                fileId: spoolId.toString(),
+                jobName: jobname,
+                jobId: jobid,
+                owner: "*",
+            };
+            const response: Buffer = await JobUtils.getSpoolFileContent(connection, options);
+            if (response) {
+                // eslint-disable-next-line @typescript-eslint/no-unsafe-return
+                return response.toString();
+            }
+        }
+        return "";
+    }
+
+    public getJclForJob(job: zowe.IJob): Promise<string> {
+        throw new Error("Get jcl is not supported.");
+    }
+
+    public submitJcl(jcl: string, internalReaderRecfm?: string, internalReaderLrecl?: string): Promise<zowe.IJob> {
+        throw new Error("Submit jcl is not supported.");
+    }
+
+    public async submitJob(jobDataSet: string): Promise<zowe.IJob> {
+        const result = this.getIJobResponse();
+        const connection = await this.ftpClient(this.checkedProfile());
+        if (connection) {
+            const transferOptions = {
+                transferType: TRANSFER_TYPE_ASCII,
+            };
+            const content = await DataSetUtils.downloadDataSet(connection, jobDataSet, transferOptions);
+            const jcl = content.toString();
+            const response: string = await JobUtils.submitJob(connection, jcl);
+            if (response) {
+                result.jobid = response;
+            }
+        }
+        return result;
+    }
+    public async deleteJob(jobname: string, jobid: string): Promise<void> {
+        const connection = await this.ftpClient(this.checkedProfile());
+        if (connection) {
+            await JobUtils.deleteJob(connection, jobid);
+        }
+    }
+
+    private getIJobResponse(): zowe.IJob {
+        return {
+            jobid: "",
+            jobname: "",
+            subsystem: "",
+            owner: "",
+            status: "",
+            type: "",
+            class: "",
+            retcode: "",
+            "step-data": [],
+            url: "",
+            "files-url": "",
+            "job-correlator": "",
+            phase: 0,
+            "phase-name": "",
+            "reason-not-running": "",
+        };
+    }
+    private getIJobFileResponse(): zowe.IJobFile {
+        return {
+            jobid: "",
+            jobname: "",
+            recfm: "",
+            "byte-count": 0,
+            "record-count": 0,
+            "job-correlator": "",
+            class: "",
+            id: 0,
+            ddname: "",
+            "records-url": "",
+            lrecl: 0,
+            subsystem: "",
+            stepname: "",
+            procstep: "",
+        };
+    }
+    private checkedProfile(): imperative.IProfileLoaded {
```

Comment:
I think `checkedProfile()` and `ftpClient()` can be moved into `AbstractFtpApi` as protected methods as well.

Reply:
Thanks @Qi! I moved them to ZoweExplorerAbstractFtpApi.ts